test/optimizer/plugins/SimplifyRulePathContextsOptimizer.test.ts

```diff
@@ -975,5 +976,42 @@ describe('optimizer', () => {
         expectedComponentOnly
       ]);
     });
+
+    it('should indent based on path rules', () => {
+      // Instance: ChainsawPatient
+      // InstanceOf: Patient
+      // Usage: #example
+      // * address
+      // * address.city = "Tokyo"
+      // * address.country = "CN"
```

Comment:
It might be nice to include the `address.period`, `address.period.start`, and `address.period.end` paths that are tested in the `'should add path rules for deeper paths'` test (on lines 106-153) in order to ensure the deeper level paths in path rules are indented correctly as well.

Reply:
That would be fine. Since the `SimplifyRulePathContextsOptimizer` was not modified in this PR, the tests are intended just to confirm that this optimizer will correctly handle the outputs of the new `AddPathRulesOptimizer`.

Reply:
I'm fine if we don't want to take this suggestion. I was just thinking it covered more of the types of things that are output by `AddPathRulesOptimizer`, but you're probably right that it isn't necessary.

Reply:
Added this in 5df2936.

Reply:
Do you think this first test is necessary to include separately from the test below it? I think the one below it is a superset of this test, and I'd be fine if we just want to keep the one.

Reply:
Let's just keep it as-is so that if (some day) the shallow one works, but the deep one gets broken, it's more obvious.